R/mboostLSS.R

```diff
@@ -122,15 +123,45 @@ mboostLSS_fit <- function(formula, data = list(), families = GaussianLSS(),
     trace <- control$trace
     control$trace <- FALSE
 
+    ## generate adequate model weights 
     w <- weights
+    
     if (is.null(weights)){
-        if (!is.list(response)) {
-            weights <- rep.int(1, NROW(response))
-        } else {
-            weights <- rep.int(1, NROW(response[[1]]))
-        }
+      if (!is.list(response)) {
+        weights <- rep.int(1, NROW(response))
+        # expand weights if the response is a matrix (functional response)
+        if(funchar == "FDboost" && !is.null(dim(response)) && !any(dim(response) == 1))
+          weights <- rep.int(weights, ncol(response))
+        
+      } else {
+        weights <- rep.int(1, NROW(response[[1]]))
+        # expand weights if the response is a matrix (functional response)
+        if(funchar == "FDboost" && !is.null(dim(response[[1]])) && !any(dim(response[[1]]) == 1))
+          weights <- rep.int(weights, ncol(response[[1]]))
+      }
     }
+    
     weights <- rescale_weights(weights)
+    
+    ## set up timeformula for FDboost 
```

Comment:
Could this be a function `check_time_formula` specified in helpers.R? This would reduce the changes in mboostLSS_fit.